src/main/java/org/takes/facets/auth/social/PsLinkedin.java

```diff
@@ -0,0 +1,217 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.takes.facets.auth.social;
+
+import com.jcabi.http.request.JdkRequest;
+import com.jcabi.http.response.JsonResponse;
+import com.jcabi.http.response.RestResponse;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import javax.json.JsonObject;
+import lombok.EqualsAndHashCode;
+import org.takes.Request;
+import org.takes.Response;
+import org.takes.facets.auth.Identity;
+import org.takes.facets.auth.Pass;
+import org.takes.misc.Href;
+import org.takes.rq.RqHref;
+
+/**
+ * Linkedin OAuth landing/callback page.
+ *
+ * <p>The class is immutable and thread-safe.
+ * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
+ * @version $Id$
+ */
+@EqualsAndHashCode(of = { "app", "key" })
+public final class PsLinkedin implements Pass {
+    /**
+     * OAUTH base url param name.
+     */
+    public static final String OAUTH = "OAUTH";
+
+    /**
+     * Linkedin API base url param name.
+     */
+    public static final String API = "API";
+
+    /**
+     * Default params.
+     */
+    private static final ConcurrentMap<String, String> DEF_PARAMS;
+
+    /**
+     * OAuth authorization code parameter.
+     */
+    private static final String CODE = "code";
+
+    /**
+     * App name.
+     */
+    private final transient String app;
+
+    /**
+     * Key.
+     */
+    private final transient String key;
+
+    /**
+     * Params.
+     */
+    private final transient ConcurrentMap<String, String> params;
+
+    static {
+        DEF_PARAMS = new ConcurrentHashMap<String, String>(2);
```

Comment:
@dmzaytsev  Maybe it would be better to use `ImmutableMap.of(PsLinkedin.OAUTH, "https://www.linkedin.com", PsLinkedin.API, "https://api.linkedin.com")`, it would be more readable and without static initializer.